TAO/tao/Transport.cpp

```diff
@@ -164,6 +163,8 @@ TAO_Transport::TAO_Transport (CORBA::ULong tag,
 #endif /* TAO_HAS_TRANSPORT_CURRENT == 1 */
   , flush_in_post_open_ (false)
 {
+  cache_map_entry_.entry_ = 0;
+  cache_map_entry_.int_id_ = 0;
```

Comment:
It appears that it would be more efficient to use 'partial_sort' instead of full sort, since we need to find only the known number of purgable entries and sort them by purging order.

I am not too sure if all platforms supported by TAO actually have std::partial_sort though, and would the default assumption be that it exists, or it doesnt (ACE_HAS_PARTIAL_SORT or ACE_LACKS_PARTIAL_SORT...)

In any case, by not using qsort, I can also introduce default constructor to HASH_MAP_ENTRY_REF class, as suggested by @jwillemsen.